<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UpstreamDialog</class>
 <widget class="QDialog" name="UpstreamDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>304</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>添加上游服务器</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="nameLabel">
       <property name="text">
        <string>服务器名称:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="nameLineEdit"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="typeLabel">
       <property name="text">
        <string>服务器类型:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="typeComboBox"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="addressLabel">
       <property name="text">
        <string>服务器地址:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="addressLineEdit">
       <property name="maxLength">
        <number>100</number>
       </property>
       <property name="frame">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="portLabel">
       <property name="text">
        <string>服务器端口:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
    <widget class="QSpinBox" name="portSpinBox">
        <property name="minimum">
        <number>1</number>
        </property>
        <property name="maximum">
        <number>65535</number>
        </property>
        <property name="value">
        <number>1080</number>
        </property>
    </widget>
    </item>
     <item row="4" column="0">
      <widget class="QLabel" name="localAddressLabel">
       <property name="text">
        <string>本地地址:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="localAddressLineEdit"/>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="localPortLabel">
       <property name="text">
        <string>本地端口:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QSpinBox" name="localPortSpinBox">
        <property name="minimum">
        <number>0</number>
        </property>
        <property name="maximum">
        <number>65535</number>
        </property>
        <property name="value">
        <number>0</number>
        </property>
    </widget>
    </item>
     <item row="6" column="0">
      <widget class="QLabel" name="encryptionMethodLabel">
       <property name="text">
        <string>加密方式:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QComboBox" name="encryptionMethodComboBox"/>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="passwordLabel">
       <property name="text">
        <string>密码:</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QLineEdit" name="passwordLineEdit">
       <property name="echoMode">
        <enum>QLineEdit::EchoMode::Password</enum>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="userNameLabel">
       <property name="text">
        <string>用户名:</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QLineEdit" name="userNameLineEdit"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>UpstreamDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>UpstreamDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
